server/world/world.go

```diff
@@ -265,6 +265,13 @@ func (w *World) SetBlock(pos cube.Pos, b Block) {
 		return
 	}
 
+	if b, ok := b.(interface {
+		Place(pos cube.Pos, w *World) bool
+	}); ok && !b.Place(pos, w) {
```

Comment:
This should probably go into PlaceBlock. SetBlock is meant to be used when performance matters (blocks aren't updated etc.), something like this can slow SetBlock down by 40-50%, interfaces are expensive.

Reply:
Moved.